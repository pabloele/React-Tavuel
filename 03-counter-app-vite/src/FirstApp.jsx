import PropTypes from "prop-types";

function FirstApp({ title, subtitle, name }) {
  if (!title) throw new Error("El title no existe");
  return (
    <>
      <div data-testid="test-title">TITULO</div>
      <h1>{title}</h1>
      <p>{subtitle + 1}</p>
      <p>{name}</p>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  title: "No hay t√≠tulo",
  subtitle: 0,
  name: "Pablo",
};

export default FirstApp;
