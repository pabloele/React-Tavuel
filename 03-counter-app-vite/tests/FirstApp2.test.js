import { getByTestId, render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
  const title = "Hola soy el título";
  const subtitle = 789654;
  test("debe hacer match con el snapshot ", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test('debe mostrar el mensaje "Hola soy el título"', () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
    // expect(screen.getByText(title)).not.toBeTruthy();
  });
  test("debe mostrar el título en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });
  test("debe mostrar el subtítulo enviado por props ", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByText(subtitle + 1)).toBeTruthy();
  });
});
