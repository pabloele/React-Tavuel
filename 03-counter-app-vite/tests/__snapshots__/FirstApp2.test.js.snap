// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot  1`] = `
<div>
  <div
    data-testid="test-title"
  >
    TITULO
  </div>
  <h1>
    Hola soy el t√≠tulo
  </h1>
  <p>
    1
  </p>
  <p>
    Pablo
  </p>
</div>
`;
